includes/AMP/Sanitization.php

```diff
@@ -0,0 +1,459 @@
+<?php
+/**
+ * Class Sanitization
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\AMP;
+
+use AMP_Allowed_Tags_Generated;
+use AMP_Content_Sanitizer;
+use AMP_DOM_Utils;
+use AMP_Layout_Sanitizer;
+use AMP_Script_Sanitizer;
+use AMP_Style_Sanitizer;
+use AMP_Tag_And_Attribute_Sanitizer;
+use AmpProject\Amp;
+use AmpProject\Attribute;
+use AmpProject\Dom\Document;
+use AmpProject\Extension;
+use AmpProject\Tag;
+use DOMElement;
+
+/**
+ * Sanitization class.
+ *
+ * @since 1.1.0
+ */
+class Sanitization {
+	/**
+	 * Sanitizes a document.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param Document $document Document instance.
+	 *
+	 * @return void
+	 */
+	public function sanitize_document( Document $document ) {
+		$sanitizers = $this->get_sanitizers();
+
+		$result = AMP_Content_Sanitizer::sanitize_document( $document, $sanitizers, [] );
+
+		$this->ensure_required_markup( $document, $result['scripts'] );
+	}
+
+	/**
+	 * Adds missing scripts.
+	 *
+	 * @since 1.1.0
```

Comment:
To facilitate updates in the future, would be good to link to the specific file revision that this is adapted from.

```suggestion
	 * @since 1.1.0
	 * @link https://github.com/ampproject/amp-wp/blob/8856284d90fc8558c30acc029becd352ae26e4e1/includes/class-amp-theme-support.php#L1543-L1770
```